src/software/new_geom/util/intersects_test.cpp

```diff
@@ -0,0 +1,414 @@
+    EXPECT_TRUE(intersectsNew(r, c));
+    EXPECT_TRUE(intersectsNew(c, r));
+}
+
+TEST(IntersectsTest, circle_outside_rectangle_not_intersecting)
+{
+    Rectangle r({-5, -2}, {3, 4});
+    Circle c({4, -5}, 2);
+    EXPECT_FALSE(intersectsNew(r, c));
+    EXPECT_FALSE(intersectsNew(c, r));
+}
+
+TEST(IntersectsTest, circles_same_origin)
+{
+    Circle c1({4, -5}, 3);
+    Circle c2({4, -5}, 2);
+    EXPECT_TRUE(intersectsNew(c1, c2));
+}
+
+TEST(IntersectsTest, circle_origin_in_other_circle)
+{
+    Circle c1({4, -5}, 3);
+    Circle c2({6, -5}, 1);
+    EXPECT_TRUE(intersectsNew(c1, c2));
+}
+
+TEST(IntersectsTest, circles_with_origins_not_overlapping)
+{
+    Circle c1({-6, 2}, 4);
+    Circle c2({4, -5}, 9);
+    EXPECT_TRUE(intersectsNew(c1, c2));
+}
+
+TEST(IntersectsTest, segment_completely_inside_circle)
+{
+    Segment s({-8, 4}, {-4, 0});
+    Circle c({-6, 2}, 4);
+    EXPECT_FALSE(intersectsNew(s, c));
+    EXPECT_FALSE(intersectsNew(c, s));
+}
+
+TEST(IntersectsTest, segment_with_points_on_circle_edge)
+{
+    Segment s({-2, 2}, {-6, 6});
+    Circle c({-6, 2}, 4);
+    EXPECT_FALSE(intersectsNew(s, c));
+    EXPECT_FALSE(intersectsNew(c, s));
+}
+
+TEST(IntersectsTest, segment_circle_intersecting)
+{
+    Segment s({-8, 2}, {-10, 6});
+    Circle c({-6, 2}, 4);
+    EXPECT_TRUE(intersectsNew(s, c));
+    EXPECT_TRUE(intersectsNew(c, s));
+}
+
+TEST(IntersectsTest, segments_overlapping)
+{
+    Segment s1({-8, 2}, {-10, 6});
+    Segment s2({-9, 4}, {-8.5, 3});
+    EXPECT_TRUE(intersectsNew(s1, s2));
+    EXPECT_TRUE(intersectsNew(s1, s2));
+}
+
+TEST(IntersectsTest, segments_end_points_overlapping)
+{
+    Segment s1({-8, 2}, {-10, 6});
+    Segment s2({-10, 6}, {10, 3});
+    EXPECT_TRUE(intersectsNew(s1, s2));
+    EXPECT_TRUE(intersectsNew(s2, s1));
+}
+
+TEST(IntersectsTest, segments_parallel)
+{
+    Segment s1({-8, 2}, {-10, 6});
+    Segment s2({-9, 6}, {-7.5, 3});
+    EXPECT_TRUE(!intersectsNew(s1, s2));
+    EXPECT_TRUE(!intersectsNew(s2, s1));
+}
+
+TEST(IntersectsTest, segments_intersecting)
+{
+    Segment s1({-8, 2}, {-10, 6});
+    Segment s2({-10, 0}, {-6, 8});
+    EXPECT_TRUE(intersectsNew(s1, s2));
+    EXPECT_TRUE(intersectsNew(s2, s1));
+}
+
+TEST(IntersectsTest, segments_far_from_each_other)
+{
+    Segment s1({-8, 2}, {-10, 6});
+    Segment s2({20000, 100000}, {40000, -20000});
+    EXPECT_FALSE(intersectsNew(s1, s2));
+    EXPECT_FALSE(intersectsNew(s2, s1));
+}
+
+TEST(IntersectsTest, close_parallel_segments_not_intersecting)
+{
+    // This is a test from a bug found
+    Segment s1({1.049, -1.049}, {1.95, -1.049});
+    Segment s2({2, -1}, {1, -1});
+    EXPECT_FALSE(intersectsNew(s1, s2));
+    EXPECT_FALSE(intersectsNew(s2, s1));
+}
+
+TEST(IntersectsTest, test_seg_crosses_seg)
+{
+    // should check for the the rare cases
+
+    for (int i = 0; i < 10; i++)
+    {
+        // generate three random points
+        Point a1(std::rand() % 200 / 100.0, std::rand() % 200 / 100.0);
```

Comment:
We really shouldn't have randomness in the tests, especially since we aren't providing a consistent see for the RNG. If there are known rare cases as mentioned in the comment above, I think it's better to break those out into their own hardcoded test cases (which also makes it more clear what's being tested).